### ISSUE TITLE
JSON data that breaks PerfCascade

### ISSUE THREAD

**Issue by Nooshu** [micmro/PerfCascade#238]
Hello,
I've been using [https://compare.sitespeed.io/](https://compare.sitespeed.io/) for some performance testing and I've run into a set of data that causes an issue in compare.  The data can be seen in [this gist](https://gist.github.com/Nooshu/2be556b73a0b3fa41f9dc3927dab7ee1). I've run it through a JSON validator and it passes fine.

A direct link to the error can be [found here](https://compare.sitespeed.io/?har1=https://sitespeed-performance-before-firefox-mobile-3g-slow.netlify.com/pages/www.gov.uk/government/speeches/spring-statement-2018-philip-hammonds-speech/data/browsertime.har&compare=1).

The error that comes back is:

```
combined-97f030e963.js:566 TypeError: Cannot read property 'toString' of undefined
    at combined-97f030e963.js:25
    at Object.n.getKeys (combined-97f030e963.js:25)
    at Object.n.makeTabs (combined-97f030e963.js:25)
    at combined-97f030e963.js:25
    at combined-97f030e963.js:25
    at Array.map (<anonymous>)
    at u (combined-97f030e963.js:25)
    at combined-97f030e963.js:25
    at Array.map (<anonymous>)
    at Object.n.transformDoc (combined-97f030e963.js:25)
fetchHar.then.catch.e @ combined-97f030e963.js:566
Promise.catch (async)
loadFilesFromURL @ combined-97f030e963.js:564
onclick @ (index):413
```
Delving into the issue a little further it looks to be a problem in the PerfCascade lib. Compare uses v2.5.2, but I've also tested with v2.5.4 and it has the same issue. The line it is failing at is this one:

```js
if (content.size !== -1 && contentLength !== content.size.toString()) {
    contentSize = content.size;
}
```
Thanks!

**Comment by soulgalore** [comment]
Hi @Nooshu thanks for creating the issue and showing the exact problem. I did a quick fix for now and I'll update compare when we got a new release.

I looked at the HAR and ther HAR file is actually not following the HAR spec and I see it's from Firefox so we should create an upstream issue.

<img width="582" alt="screenshot 2018-10-21 at 22 48 07" src="https://user-images.githubusercontent.com/540757/47272261-6e22b300-d583-11e8-914e-4d6207f24008.png">

We use https://github.com/devtools-html/har-export-trigger to get the HAR from Firefox, but I guess then the real upstream issue is in Firefox. If you can export the HAR directly from FF and either create an upstream issue in https://bugzilla.mozilla.org/ or ping me and I can do it.

Best
Peter

**Comment by Nooshu** [comment]
Hi @soulgalore,

Okay that is interesting. I've just tried exporting a HAR file from FF from the [page](https://www.gov.uk/government/speeches/spring-statement-2018-philip-hammonds-speech) using the latest version of FF I have (62.0.3) and [the resulting file](https://gist.github.com/Nooshu/955b5b00adefc49e378fa251b0a67940) works in compare and HAR Viewer validates correctly. I throttled the connection to 3G Slow while running SiteSpeed in a docker container. Other than that all other settings were standard (and all other HAR's using the same setup exported fine).

I'm wondering if something went wrong in the export for that particular run of tests? As at the moment I'm unable to recreate the issue.

I'd say that unless it is possible to reproduce this it may actually be a non-issue.
